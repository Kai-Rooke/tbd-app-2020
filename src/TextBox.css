.textbox {
    margin: 1.5% 3% 1.5% 3%;
}

.textbox input {
    background-color: white;
    color: black;
    padding: 10px;
    border: 2px solid black;
    margin-left: 1.5%;
}

.textbox input:focus {
    outline: none !important;
    border: 2px solid #31b7bc;
}