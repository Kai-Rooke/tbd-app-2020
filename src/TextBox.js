import React from 'react';
import PropTypes from 'prop-types';

export const TextBox = ({backgroundColor, title, size, id, maxlength, minlength, ...props}) => {
return (
    <div>
        <label for={id}>{title}</label>
        <input
        type="text"
        id= {id}
        maxlength = {maxlength}
        minlength = {minlength}
        />
    </div>
);
};

TextBox.propTypes = {
    backgroundColor: PropTypes.string,
        /** What background colour to use */
    title: PropTypes.string.isRequired,
        /** Input contents */
    size: PropTypes.oneOf(['small', 'medium', 'large']),
        /** How large should the input be? */
    maxlength: PropTypes.number,
    //Maximum input length
    minlength: PropTypes.number
    //Minimum input length
};

TextBox.defaultProps = {
    backgroundColor: null,
    size: 'small',
    maxlength: 30,
    minlength: 0
};