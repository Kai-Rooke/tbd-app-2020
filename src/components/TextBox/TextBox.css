.textbox {
    margin: 0% 20% 10px 1%;
    max-width: 150px;
}


.textbox input {
    background-color: white;
    color: black;
    padding: 10px;
    border: 2px solid black;
    margin-left: 1.5%;
}

.textbox input:focus {
    outline: none !important;
    border: 2px solid #31b7bc;
}
@media screen and (max-width: 650px) {
    .textbox input {
        max-width: 120%!important;
    }
    .textbox  {
        margin-right: 20%!important;
    }
  }