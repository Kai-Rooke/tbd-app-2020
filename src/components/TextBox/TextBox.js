import React from 'react';
import './TextBox.css'
import PropTypes from 'prop-types';

const TextBox = ({backgroundColor, name, title, size, id, maxLength, minLength, placeholder, type, onChange, min, max, wrapID, ...props}) => {
return (
    <div className="textbox" id={wrapID}>
        <label htmlFor={id}>{title}</label>
        <input
        name={name}
        onChange={onChange}
        placeholder={placeholder}
        id= {id}
        maxLength = {maxLength}
        minLength = {minLength}
        size={size}
        type={type}
        min={min}
        max={max}
        />
    </div>
);
};

TextBox.propTypes = {
    backgroundColor: PropTypes.string,
    title: PropTypes.string,
    size: PropTypes.number,
    maxLength: PropTypes.number,
    minLength: PropTypes.number,
    placeholder: PropTypes.string,
    type: PropTypes.oneOf(['text', 'password', 'email', 'tel', 'number', 'date']),
    min: PropTypes.number,
    max: PropTypes.number,
    wrapID: PropTypes.string
};

TextBox.defaultProps = {
    backgroundColor: null,
    size: 20,
    maxLength: 30,
    minLength: 0,
    placeholder: '',
    type: "text"
};

/* Example Use

<TextBox 
    title="First Name" 
    placeholder="Enter your firstname" 
    minLength="5" 
    maxLength="30" 
    id="input1" 
    type="text" 
/>

*/

export default TextBox;
