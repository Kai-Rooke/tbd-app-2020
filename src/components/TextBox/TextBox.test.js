import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import TextBox from './TextBox';

//cleans up 
afterEach(cleanup);

it("Renders TextBox", () => {
    const { asFragment } = render(<TextBox   title="First Name" 
    placeholder="Enter your firstname" 
    minLength="5" 
    maxLength="30" 
    id="input1" 
    type="text" />);
    expect(asFragment()).toMatchSnapshot();
});
