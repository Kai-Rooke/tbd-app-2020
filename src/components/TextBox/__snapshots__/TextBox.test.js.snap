// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders TextBox 1`] = `
<DocumentFragment>
  <div
    class="mobiletextbox"
  >
    <label
      for="input1"
    >
      First Name
    </label>
    <input
      id="input1"
      maxlength="30"
      minlength="5"
      placeholder="Enter your firstname"
      size="20"
      type="text"
    />
  </div>
</DocumentFragment>
`;
